app/scripts/lib/local-store.js

```diff
@@ -88,3 +88,17 @@ module.exports = class ExtensionStore {
 function isEmpty (obj) {
   return Object.keys(obj).length === 0
 }
+
+/**
+ * Returns an Error if extension.runtime.lastError is present
+ * this is a workaround for the non-standard error object thats used
+ * @returns {Error}
+ */
+function checkForError () {
+  const lastError = extension.runtime.lastError
```

Comment:
Ideally we would never call `extension` directly from within the `metamask-controller`, we should only call methods passed into its constructor, which would ideally allow portability between platforms. But who am I kidding, we already rewrote most of that controller for mobile anyways.